### PR TITLE
tiflash: update expression list supported by TiFlash in 4.0

### PR DESCRIPTION
**Pull request by windtalker**
### What is changed, added or deleted? (Required)

update Expression list that supported by TiFlash
### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by yikeke** [comment]
You also need to update the `master` branch. The best practice is that you push commits to the `master` branch and use the `sre-bot` to automatically cherry-pick the same commits to the `release-4.0` branch.

Can you close this PR and open another PR in `master` branch? @windtalker

**Comment by windtalker** [comment]
> You also need to update the `master` branch. The best practice is that you push commits to the `master` branch and use the `sre-bot` to automatically cherry-pick the same commits to the `release-4.0` branch.
> 
> Can you close this PR and open another PR in `master` branch? @windtalker

This is pr release-related, the change in master is different from the change in release branch, So I have to update master/release-4.0/release-3.1 separately.

**Comment by yikeke** [comment]
> > You also need to update the `master` branch. The best practice is that you push commits to the `master` branch and use the `sre-bot` to automatically cherry-pick the same commits to the `release-4.0` branch.
> > Can you close this PR and open another PR in `master` branch? @windtalker
> 
> This pr is release-related, the change in master is different from the change in release branch, So I have to update master/release-4.0/release-3.1 separately.

If the `master` changes are different from the `release-4.0` changes, then you are right. After you update the `master` branch, please involve me as the reviewer.

File: reference/tiflash/use-tiflash.md [hunk 1]
```diff
@@ -191,9 +191,14 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
+>
+> TiFlash 在开启[新的 Collation 框架](/reference/sql/characterset-and-collation.md#新框架下的-collation-支持)后不支持任何表达式的下推。
+
 TiFlash 主要支持谓词、聚合下推计算，下推的计算可以帮助 TiDB 进行分布式加速。暂不支持的计算类型主要是表连接和 DISTINCT COUNT，会在后续版本逐步优化。
 
-目前 TiFlash 支持了有限的常用表达式下推，想了解具体下推的表达式，请参考[该文件](https://github.com/pingcap/tidb/blob/release-3.1/expression/expression.go#L409)。
+目前 TiFlash 支持了有限的常用表达式下推，想了解具体下推的表达式，请参考[该文件](https://github.com/pingcap/tidb/blob/v4.0.0-rc/expression/expression.go#L979)。
+此外，目前 TiFlash 对于所有包含 Duration 和 JSON 的表达式均不能下推。
 
 例如在聚合函数或者 WHERE 条件中包含了不在上述列表中的表达式，聚合或者相关的谓词过滤会无法进行下推。
```
Comment (yikeke) [inline reference/tiflash/use-tiflash.md:203]:
```suggestion
目前 TiFlash 支持了有限的常用表达式下推，支持下推的表达式可参考[该文件](https://github.com/pingcap/tidb/blob/v3.1.0/expression/expression.go#L558)。

目前 TiFlash 不支持下推的情况包括：

- 所有包含 Duration 和 JSON 的表达式均不能下推
- 在聚合函数或者 WHERE 条件中包含了不在[该文件](https://github.com/pingcap/tidb/blob/v3.1.0/expression/expression.go#L558)列表中的表达式，聚合或者相关的谓词过滤均不能下推
```
To refine the format.

File: reference/tiflash/use-tiflash.md [hunk 2]
```diff
@@ -191,10 +191,17 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
...
 ## TiFlash 支持的计算下推
 
+> **注意：**
+>
+> TiFlash 在开启[新的 Collation 框架](/reference/sql/characterset-and-collation.md#新框架下的-collation-支持)后不支持任何表达式的下推。
+
 TiFlash 主要支持谓词、聚合下推计算，下推的计算可以帮助 TiDB 进行分布式加速。暂不支持的计算类型主要是表连接和 DISTINCT COUNT，会在后续版本逐步优化。
 
-目前 TiFlash 支持了有限的常用表达式下推，想了解具体下推的表达式，请参考[该文件](https://github.com/pingcap/tidb/blob/release-3.1/expression/expression.go#L409)。
+目前 TiFlash 支持了有限的常用表达式下推，支持下推的表达式可参考[该文件](https://github.com/pingcap/tidb/blob/v4.0.0-rc/expression/expression.go#L979)。
```
Comment (ilovesoup) [inline reference/tiflash/use-tiflash.md:200]: 这个记得后续要update。。给自己加个todo吧？

**Pull request merged**